### ISSUE TITLE
The NF does not handle correctly Modelica.Math.Nonlinear.SolveOneNonlinearEquation

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#7990]
### Description

Please consider the following model

```Modelica
model TestMoistAirInverse
  package Medium = Modelica.Media.Air.MoistAir;
  Medium.ThermodynamicState state = Medium.setState_phX(1e5, 98000, {0.01});
  annotation(uses(Modelica(version="4.0.0")));
end TestMoistAirInverse;
```
The OF handles it nicely. The NF gives:
```
[1] 10:21:50 Translation Error
[Modelica.Math.Nonlinear: 413:5-413:46]: Output parameter y was not assigned a value

```
### Expected Behavior

The NF should also handle this model with the same results.

**Comment by casella** [comment]
@perost, if this is an easy fix, it would be nice to have in 1.19.0. Thanks!

**Comment by perost** [comment]
Fixed in #8302. The issue was that the function evaluation in the NF didn't handle function partial application properly (the OF doesn't either, but it just fails and skips evaluating it instead). With the NF the whole model is now flattened to just:
```Modelica
class TestMoistAirInverse_total
  Real state.p(quantity = "Pressure", unit = "Pa", displayUnit = "bar", min = 0.0, max = 100000000.0, start = 100000.0, nominal = 100000.0) = 100000.0 "Absolute pressure of medium";
  Real state.T(quantity = "ThermodynamicTemperature", unit = "K", displayUnit = "degC", min = 190.0, max = 647.0, start = 288.15, nominal = 300.0) = 345.1276781054744 "Temperature of medium";
  Real state.X[1](quantity = "MassFraction", unit = "kg/kg", min = 0.0, max = 1.0, start = 0.01, nominal = 0.1) "Mass fractions (= (component mass)/total mass  m_i/m)";
  Real state.X[2](quantity = "MassFraction", unit = "kg/kg", min = 0.0, max = 1.0, start = 0.99, nominal = 0.1) "Mass fractions (= (component mass)/total mass  m_i/m)";
equation
  state.X = {0.01, 0.99};
end TestMoistAirInverse_total;
```
I will keep this issue open a while longer though, I need to add a test case for it too but I'll do that next week.

**Comment by perost** [comment]
> I will keep this issue open a while longer though, I need to add a test case for it too but I'll do that next week.

Test cases where added in #8311, but seems I forgot to close this.
